nautobot/extras/models/groups.py

```diff
@@ -165,8 +165,15 @@ def _map_filter_fields(self):
         # Get dynamic group filter field mappings (if any)
         dynamic_group_filter_fields = getattr(self.model, "dynamic_group_filter_fields", {})
 
-        # Model form fields that aren't on the filter form
-        missing_fields = set(modelform_fields).difference(filterform_fields)
+        # Whether or not to add missing form fields that aren't on the filter form.
+        skip_missing_fields = getattr(self.model, "dynamic_group_skip_missing_fields", False)
+
+        # Model form fields that aren't on the filter form.
+        if not skip_missing_fields:
+            missing_fields = set(modelform_fields).difference(filterform_fields)
+        else:
+            logger.debug("Will not add missing form fields due to model meta option.")
```

Comment:
Maybe something like:

```suggestion
            logger.debug("Will not add missing form fields due to model %s meta option.", self.model)
```